app/main.py

```diff
@@ -1,2 +1,79 @@
+from typing import Any, Hashable
+
+
 class Dictionary:
-    pass
+
+    class Node:
+        def __init__(self, key: Hashable, value: Any) -> None:
+            self.key = key
+            self.value = value
+            self.hash = hash(key)
+
+    def __init__(self) -> None:
+        self.capacity = 8
+        self.size = 0
+        self.load_factor = 2 / 3
+        self.hash_table = [None] * self.capacity
+
+    def __len__(self) -> int:
+        return self.size
+
+    def index(self, key: Hashable) -> int:
+        return hash(key) % self.capacity
+
+    def resize(self) -> None:
+        new_capacity = self.capacity * 2
+        new_hash_table = [None] * new_capacity
+
+        for bucket in self.hash_table:
+            if bucket is not None:
+                for node in bucket:
+                    new_index = node.hash % new_capacity
+                    if new_hash_table[new_index] is None:
+                        new_hash_table[new_index] = []
+                    new_hash_table[new_index].append(node)
+
+        self.hash_table = new_hash_table
+        self.capacity = new_capacity
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        if self.size / self.capacity > self.load_factor:
```

Comment:
The condition `self.size / self.capacity > self.load_factor` uses float division. Ensure that `self.load_factor` is intended to be a float. If integer division was intended, adjust the load factor calculation accordingly.